controllers/machinedeployment_sync_test.go

```diff
@@ -0,0 +1,223 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha2"
+	capierrors "sigs.k8s.io/cluster-api/errors"
+)
+
+func TestMachineDeploymentSyncStatus(t *testing.T) {
+	replicas := int32(2)
+	msStatusError := capierrors.MachineSetStatusError("some failure")
+
+	var statustests = []struct {
+		desc           string
+		machineSets    []*clusterv1.MachineSet
+		newMachineSet  *clusterv1.MachineSet
+		deployment     *clusterv1.MachineDeployment
+		expectedStatus clusterv1.MachineDeploymentStatus
+	}{
+		{
+			desc: "all machines are running",
+			machineSets: []*clusterv1.MachineSet{{
+				Spec: clusterv1.MachineSetSpec{
+					Replicas: &replicas, // 2 replicas
```

Comment:
```suggestion
					Replicas: pointer.Int32Ptr(2),
```

from k8s.io/utils/pointer